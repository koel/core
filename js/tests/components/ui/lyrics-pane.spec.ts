import Lyrics from '@/components/ui/lyrics-pane.vue'
import factory from '@/tests/factory'
import { shallow } from '@/tests/adapter'

describe('components/ui/lyrics', () => {
  it('displays lyrics if the song has lyrics', () => {
    expect(shallow(Lyrics, {
      propsData: {
        song: factory('song', { lyrics: 'Foo and bar' })
      }
    })).toMatchSnapshot()
  })

  it('displays a fallback message if the song has no lyrics', () => {
    expect(shallow(Lyrics, {
      propsData: {
        song: factory('song', { lyrics: '' })
      }
    })).toMatchSnapshot()
  })
})
