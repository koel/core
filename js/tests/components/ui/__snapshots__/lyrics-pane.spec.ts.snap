// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ui/lyrics displays a fallback message if the song has no lyrics 1`] = `
<article id="lyrics">
  <div class="content">
    <div style="display: none;">
      <div></div>
      <textzoomer-stub></textzoomer-stub>
    </div>
    <p class="none"><span>No lyrics available. Are you listening to Bach?</span></p>
  </div>
</article>
`;

exports[`components/ui/lyrics displays lyrics if the song has lyrics 1`] = `
<article id="lyrics">
  <div class="content">
    <div>
      <div>Foo and bar</div>
      <textzoomer-stub></textzoomer-stub>
    </div>
    <!---->
  </div>
</article>
`;
